root/etc/s6-overlay/s6-rc.d/init-permissions-config/run

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/with-contenv bash
+# shellcheck shell=bash
+
+for file in /config/log/nginx/access.log /config/log/nginx/error.log; do
+    if [[ ! -f "${file}" ]]; then
+        touch "${file}"
+    fi
+done
+
+# permissions
+lsiown -R abc:abc \
+    /config
```

Comment:
I am still hesitant on the recursive chown on `/config` being here. I know we used to do it in the nginx base, but what would be the reason we need it here (outside of restoring the previous behavior)? I could maybe see doing a recursive chown on `/config/www` but that still makes me think permissions should maybe be up to the user. The other folders created in `/config` include `keys`, `log` (which sparked this), `nginx`, and `php` all of which maybe would be better to be handled by doing chowns in the nginx base. The `www` folder is either user-land or populated by the downstream images, and I can agree if there does need to be a recursive chown it can be added to downstream images, but I'm not positive if it's 100% the right option in this image (or swag) as they are the two most likely to have a user apply different permissions with some intention.

Reply:
I reviewed the nginx base, and here is what we're doing
https://github.com/linuxserver/docker-baseimage-alpine-nginx/blob/a05dfb3ef3424235465cc5d30e91d471a0c870af/root/etc/s6-overlay/s6-rc.d/init-permissions/run#L9-L16
Could we not just move the logs chown up to the recursive set and add the pre-creation of the files to the nginx base?

Reply:
It's up to you. You're the maintainer. I'm just letting you know that there are currently two issues, one affects all nginx based images where the logs in a fresh container are root owned and logrotate is busted as a result. The other affects the nginx image where if the user changes the PUID/PGID after creation, a bunch of files have incorrect ownership.

Reply:
> It's up to you. You're the maintainer. I'm just letting you know that there are currently two issues, one affects all nginx based images where the logs in a fresh container are root owned and logrotate is busted as a result. The other affects the nginx image where if the user changes the PUID/PGID after creation, a bunch of files have incorrect ownership.

Does the second issue not affect all nginx based images? I think that's the part I'm misunderstanding.

Reply:
Wait... I get it now.

The second issue technically does affect all nginx based images that don't already set permissions in `/config/www` but in the case of this image and swag I'm not sure if the best option is to adjust it for the user or to have the user adjust the permissions on their own. If the user changed PUID/PGID that's an intentional change on their part (even if they don't know what it can cause). For those that do know what it does and want more specific control over permissions if we add a recursive chown on `/config/www` it limits that ability. For all other images (where the idea is more to run a specific app) I do think the recursive chown on `/config/www` is probably fine.

I'll prep a PR for the nginx base to fix the logrotate issue and chat a bit in discord about the secondary issue. It'll kind of come down to support efforts I think.

Reply:
Most if not all Nginx based images do a recursive chown in their init (including SWAG). Nginx image does not because it was relying on the baseimage's recursive chown which got removed a few months back.

Reply:
I didn't go look to see it in swag, but yeah I see it now. I guess at this point if we merge https://github.com/linuxserver/docker-baseimage-alpine-nginx/pull/151 we can remove the log precreate in this PR and keep the recursive chown and call it a day.